### PR TITLE
Created a Model to Update the user  

### PR DESCRIPTION
**Pull request by Siddhartha-bethi**

### CODE REVIEW

File: backend/utils/logger.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 var log4js = require('log4js');
-const logLevel = process.env.LOG_LEVEL || 'info';
+const logLevel = process.env.LOG_LEVEL || 'debug';
```
Comment (gsvamsiBZ) [inline backend/utils/logger.js:2]: revert this back

File: backend/routes/allRoutes.js [hunk 2]
```diff
@@ -7,5 +7,6 @@ router.get("/getAllRecords",truecallerUserLib.getAllRecords)
 router.get("/getRecordByNumber",truecallerUserLib.getRecordByNumber)
 router.get('/deleteTruecallerUser',truecallerUserLib.deleteTruecallerUser)
 router.post('/insertTruecallerUser',truecallerUserLib.insertTruecallerUser)
+router.post("/findandupdate",truecallerUserLib.findandupdate)
```
Comment (gsvamsiBZ) [inline backend/routes/allRoutes.js:10]: use camel casing

File: frontend/src/Component/homepage.js [hunk 3]
```diff
@@ -29,16 +39,98 @@ function HomePage() {
       console.log(error);
     })
   }
+  async function add(e) {
```
Comment (gsvamsiBZ) [inline frontend/src/Component/homepage.js:42]: add a comment on what this function is

File: frontend/src/Component/homepage.js [hunk 4]
```diff
@@ -29,16 +39,98 @@ function HomePage() {
...
-      <td >{element.location}</td>
+    <tr key={element.phone} onClick={() => { show(element) }}>
+      <td>{element.name}</td>
+      <td>{element.phone}</td>
+      <td>{element.email}</td>
+      <td>{element.location}</td>
     </tr>
   ));
-
+  function validatemobile(number) {
```
Comment (gsvamsiBZ) [inline frontend/src/Component/homepage.js:113]: camel casing

File: frontend/src/Component/homepage.js [hunk 5]
```diff
@@ -29,16 +39,98 @@ function HomePage() {
...
+      return false
+    }
+    for (let x of number) {
+      if (isNaN(x)) {
+        return false
+      }
+    }
+    return true
+  }
+  function validateemail(email) {
```
Comment (gsvamsiBZ) [inline frontend/src/Component/homepage.js:124]: camel casing

**Review by gsvamsiBZ — CHANGES_REQUESTED** [review]

File: frontend/src/Component/homepage.js [hunk 6]
```diff
@@ -1,22 +1,32 @@
 import axios from 'axios';
 import React, { useEffect, useState } from "react";
-import { Table, Container, Input, Group } from '@mantine/core';
+import { Table, Modal, Container, Stack } from '@mantine/core';
+import { TextInput, Button, Group, Box } from '@mantine/core';
```
Comment (gsvamsiBZ) [inline frontend/src/Component/homepage.js:4]: Make it as 1 import

File: frontend/src/Component/homepage.js [hunk 7]
```diff
@@ -1,22 +1,32 @@
...
   let [limit, setPage] = useState(1);
   let [query, SetQuery] = useState("");
-
+  const [data, setData] = useState([]);
+  const [opened, setOpened] = useState(false);
+  const [details, setDetails] = useState({})
+  const myname = useRef();
+  const myemail = useRef();
+  const myphone = useRef();
+  const mylocation = useRef()
```
Comment (gsvamsiBZ) [inline frontend/src/Component/homepage.js:25]: make variable names as camelcasing

File: frontend/src/Component/homepage.js [hunk 8]
```diff
@@ -30,15 +40,91 @@ function HomePage() {
...
+        getAllRecords()
+        setOpened(false)
+      }
+      catch (err) {
+        console.log(err)
+        if (err?.response?.status == 409) {
+          showNotification({
+            title: "Error",
+            message: "Phone Number already exists",
+            autoClose: 4000,
```
Comment (gsvamsiBZ) [inline frontend/src/Component/homepage.js:76]: store autoclose timeout time in a const variable and use it everywhere

**Review by gsvamsiBZ — CHANGES_REQUESTED** [review]

**Comment by Siddhartha-bethi** [comment]
image
image

**Review by gsvamsiBZ — APPROVED** [review]

**Comment by Siddhartha-bethi** [comment]
image
image

**Comment by Siddhartha-bethi** [comment]
image
image

File: backend/lib/truecallerUserLib.js [hunk 9]
```diff
@@ -80,3 +80,35 @@ module.exports.getRecordByNumber = async (req, res) => {
...
+      duplicateRecord = await asyncDbLib.getOneDocumentByFilter(truecallerUserModel, { phone: req?.body?.phone })
+      logger.debug("duplicate Record is ", duplicateRecord)
+    }
+    if (duplicateRecord) {
+      res.status(409).json(duplicateRecord);
+    }
+    else {
+      try {
+        const result = await asyncDbLib.findOneDocumentByFilterAndUpdate(truecallerUserModel, filter, data)
+        logger.debug("result is ", result)
```
Comment (Karthik-184) [inline backend/lib/truecallerUserLib.js:101]: won't it take automatic space while printing?

**Review by Karthik-184 — CHANGES_REQUESTED** [review]

File: backend/lib/truecallerUserLib.js (hunk 9 above)
Comment (Siddhartha-bethi) [reply backend/lib/truecallerUserLib.js:101]: Its user wish 
Not a standard

**Comment by gsvamsiBZ** [comment]
Resolve conflicts

**Review by Karthik-184 — APPROVED** [review]

**Review by gsvamsiBZ — CHANGES_REQUESTED** [review]
backend package and package-lock are deleting mongoose paginate change that

**Pull request merged**